syntax = "proto3";

package nameserver;

option cc_generic_services = true;

message HeartBeatRequest
{
    string name = 1;
    string host = 4;
    bool has_change = 2;
    repeated string change_files = 3;
    repeated bool add = 5;
}

message HeartBeatResponse
{
    bool success = 1;
}


service NameService {
    rpc HeartBeat(HeartBeatRequest) returns(HeartBeatResponse);
}